From 475c07a9d2b999b16d76d31f8da30edfe4f0f688 Mon Sep 17 00:00:00 2001
From: Jacob Stiffler <j-stiffler@ti.com>
Date: Mon, 25 Feb 2019 09:06:33 -0500
Subject: [PATCH] HACK: board-cfg: disable MSMC cache

Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 board-cfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board-cfg.c b/board-cfg.c
index 8774268..bdc8236 100644
--- a/board-cfg.c
+++ b/board-cfg.c
@@ -69,7 +69,7 @@ const struct k3_boardcfg am65_boardcfg_data = {
 			.magic = BOARDCFG_MSMC_MAGIC_NUM,
 			.size = sizeof(struct boardcfg_msmc),
 		},
-		.msmc_cache_size = 0x10,
+		.msmc_cache_size = 0x0,
 	},
 
 	/* boardcfg_dbg_cfg */
-- 
2.7.4

