diff --git a/qt5_apps/qt5_deform/qt5_deform.desktop b/qt5_apps/qt5_deform/qt5_deform.desktop
index 9ad1d6f..ffe8e8a 100644
--- a/qt5_apps/qt5_deform/qt5_deform.desktop
+++ b/qt5_apps/qt5_deform/qt5_deform.desktop
@@ -4,7 +4,7 @@
 Name=Deform
 GenericName=Demo App
 Icon=/usr/share/matrix-gui-2.0/apps/images/qt-icon.png
-Exec=/usr/share/qt5/examples/widgets/painting/deform/deform -small-screen
+Exec=/usr/share/qt5/examples/widgets/painting/deform/deform
 Type=Application
 Categories=Qt5
 ProgramType=gui
