diff --git a/move_base/src/move_base.cpp b/move_base/src/move_base.cpp
index c608d37..43e0e8c 100644
--- a/move_base/src/move_base.cpp
+++ b/move_base/src/move_base.cpp
@@ -3,6 +3,7 @@
 * Software License Agreement (BSD License)
 *
 *  Copyright (c) 2008, Willow Garage, Inc.
+*  Copyright (c) 2017, Texas Instruments Incorporated
 *  All rights reserved.
 *
 *  Redistribution and use in source and binary forms, with or without
@@ -662,6 +663,9 @@ namespace move_base {
       controller_costmap_ros_->start();
     }
 
+    // Perform initial rotation
+    recovery_behaviors_[recovery_index_]->runBehavior();
+
     //we want to make sure that we reset the last time we had a valid plan and control
     last_valid_control_ = ros::Time::now();
     last_valid_plan_ = ros::Time::now();
