diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index c64ecb2..94ee28e 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -146,7 +146,7 @@ static void
 gst_h264_parse_init (GstH264Parse * h264parse)
 {
   h264parse->frame_out = gst_adapter_new ();
-  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (h264parse), FALSE);
+  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (h264parse), TRUE);
 }
 
 
diff --git a/gst/videoparsers/gstmpeg4videoparse.c b/gst/videoparsers/gstmpeg4videoparse.c
index 333db5d..391f122 100644
--- a/gst/videoparsers/gstmpeg4videoparse.c
+++ b/gst/videoparsers/gstmpeg4videoparse.c
@@ -186,7 +186,7 @@ gst_mpeg4vparse_init (GstMpeg4VParse * parse)
   parse->interval = DEFAULT_CONFIG_INTERVAL;
   parse->last_report = GST_CLOCK_TIME_NONE;
 
-  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), FALSE);
+  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), TRUE);
 }
 
 static void
diff --git a/gst/videoparsers/gstmpegvideoparse.c b/gst/videoparsers/gstmpegvideoparse.c
index 1feb3c5..c647a68 100644
--- a/gst/videoparsers/gstmpegvideoparse.c
+++ b/gst/videoparsers/gstmpegvideoparse.c
@@ -175,7 +175,7 @@ gst_mpegv_parse_init (GstMpegvParse * parse)
 {
   parse->config_flags = FLAG_NONE;
 
-  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), FALSE);
+  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), TRUE);
 }
 
 static void
