From 509b447f3097656b13b26aa99618ad1c281da559 Mon Sep 17 00:00:00 2001
From: Krunal Bhargav <k-bhargav@ti.com>
Date: Wed, 25 May 2022 06:30:54 -0500
Subject: [PATCH 3/3] HMI: Fix background rectangle size

Signed-off-by: Krunal Bhargav <k-bhargav@ti.com>
---
 main.qml | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/main.qml b/main.qml
index c44017f..b00275f 100644
--- a/main.qml
+++ b/main.qml
@@ -18,7 +18,8 @@ Window {
         Image {
             id: backgroundimage
             source:"images/Background.png"
-            fillMode: Image.PreserveAspectFit
+            width: parent.width
+            height: parent.height
         }
         Image {
             id: motorimage
@@ -143,8 +144,8 @@ Window {
                 anchors.top: parent.top
                 anchors.topMargin: parent.height * 0.75
                 anchors.left: parent.left
-                anchors.leftMargin: parent.width * 0.36
-                width: parent.width * 0.26
+                anchors.leftMargin: parent.width * 0.33
+                width: parent.width * 0.31
                 height: parent.height * 0.12
                 Text {
                     id: motor1text
@@ -206,7 +207,7 @@ Window {
                     fillMode: Image.PreserveAspectFit
                     width: parent.height
                     height: parent.height
-                    x: parent.width * 0.68
+                    x: parent.width * 0.72
                     MouseArea {
                         anchors.fill: parent
                         onClicked: {
@@ -343,8 +344,8 @@ Window {
                 anchors.top: parent.top
                 anchors.topMargin: parent.height * 0.75
                 anchors.left: parent.left
-                anchors.leftMargin: parent.width * 0.36
-                width: parent.width * 0.26
+                anchors.leftMargin: parent.width * 0.33
+                width: parent.width * 0.31
                 height: parent.height * 0.12
                 Text {
                     id: motor2text
@@ -405,7 +406,7 @@ Window {
                     fillMode: Image.PreserveAspectFit
                     width: parent.height
                     height: parent.height
-                    x: parent.width * 0.68
+                    x: parent.width * 0.72
                     MouseArea {
                         anchors.fill: parent
                         onClicked: {
-- 
2.24.3

