diff --git a/drivers/media/platform/ti-vpe/vip.c b/drivers/media/platform/ti-vpe/vip.c
index 5f84d42..ab44703 100644
--- a/drivers/media/platform/ti-vpe/vip.c
+++ b/drivers/media/platform/ti-vpe/vip.c
@@ -2870,12 +2870,6 @@  static int vip_probe(struct platform_device *pdev)
 	vip_shared_set_clock_enable(shared, 1);
 	vip_top_vpdma_reset(shared);
 
-	shared->vpdma = vpdma_create(pdev, vip_vpdma_fw_cb);
-	if (!shared->vpdma) {
-		dev_err(&pdev->dev, "Creating VPDMA failed");
-		goto do_iounmap;
-	}
-
 	list_add_tail(&shared->list, &vip_shared_list);
 	platform_set_drvdata(pdev, shared);
 	atomic_set(&shared->devs_allocated, 0);
@@ -2938,6 +2932,12 @@  static int vip_probe(struct platform_device *pdev)
 		vip_set_slice_path(dev, VIP_MULTI_CHANNEL_DATA_SELECT);
 	}
 
+	shared->vpdma = vpdma_create(pdev, vip_vpdma_fw_cb);
+	if (!shared->vpdma) {
+		dev_err(&pdev->dev, "Creating VPDMA failed");
+		goto dev_unreg;
+	}
+
 	return 0;
 
 dev_unreg:
