From 885ce87169ecee1b96dd2996d40293303404e4b8 Mon Sep 17 00:00:00 2001
From: Sai Sree Kartheek Adivi <s-adivi@ti.com>
Date: Fri, 24 Mar 2023 19:39:08 +0530
Subject: [PATCH] create-sdcard.sh: Copy all sysfw itb binaries to boot
 partition

Signed-off-by: Sai Sree Kartheek Adivi <s-adivi@ti.com>
---
 create-sdcard.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/create-sdcard.sh b/create-sdcard.sh
index 5c3490a..f9bce46 100644
--- a/create-sdcard.sh
+++ b/create-sdcard.sh
@@ -777,7 +777,7 @@ if [ $FILEPATHOPTION -eq 1 ] ; then
 	MLO=`ls $BOOTFILEPATH | grep MLO | awk {'print $1'}`
 	SPL_A53=`ls $BOOTFILEPATH | grep tispl | awk {'print $1'}`
 	SPL_R5=`ls $BOOTFILEPATH | grep tiboot3 | awk {'print $1'}`
-	SYSFW=`ls $BOOTFILEPATH | grep sysfw | awk {'print $1'}`
+	SYSFW=`ls $BOOTFILEPATH | grep sysfw.itb | awk {'print $1'}`
 	KERNELIMAGE=`ls $BOOTFILEPATH | grep [uz]Image | awk {'print $1'}`
 	BOOTIMG=`ls $BOOTFILEPATH | grep u-boot | grep .img | awk {'print $1'}`
 	BOOTBIN=`ls $BOOTFILEPATH | grep u-boot | grep .bin | awk {'print $1'}`
@@ -1016,7 +1016,7 @@ EOM
 		MLO=`ls $BOOTFILEPATH | grep MLO | awk {'print $1'}`
 		SPL_A53=`ls $BOOTFILEPATH | grep tispl | awk {'print $1'}`
 		SPL_R5=`ls $BOOTFILEPATH | grep tiboot3 | awk {'print $1'}`
-		SYSFW=`ls $BOOTFILEPATH | grep sysfw | awk {'print $1'}`
+		SYSFW=`ls $BOOTFILEPATH | grep sysfw.itb | awk {'print $1'}`
 		BOOTIMG=`ls $BOOTFILEPATH | grep u-boot | grep .img | awk {'print $1'}`
 		BOOTBIN=`ls $BOOTFILEPATH | grep u-boot | grep .bin | awk {'print $1'}`
 		BOOTUENV=`ls $BOOTFILEPATH | grep uEnv.txt | awk {'print $1'}`
@@ -1102,7 +1102,7 @@ if [ $BOOTPATHOPTION -eq 1 ] ; then
 
 	#copy boot files out of board support
 	if [ "$SYSFW" != "" ] ; then
-		cp $BOOTFILEPATH/$SYSFW $PATH_TO_SDBOOT/sysfw.itb
+		cp $BOOTFILEPATH/sysfw*.itb $PATH_TO_SDBOOT/
 		echo "sysfw.itb copied"
 
 		echo ""
-- 
2.34.1

