From d39697be3cdfe082dc6d3c12d6c3b78055393ff7 Mon Sep 17 00:00:00 2001
From: Lucas Weaver <l-weaver@ti.com>
Date: Tue, 15 Sep 2020 01:43:53 +0530
Subject: [PATCH] [PATCH] setup-uboot-env-am65x: Adding sudo permissions

---
 setup-uboot-env-am65x.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup-uboot-env-am65x.sh b/setup-uboot-env-am65x.sh
index b6e043b..30ce932 100644
--- a/setup-uboot-env-am65x.sh
+++ b/setup-uboot-env-am65x.sh
@@ -219,7 +219,7 @@ if [ "$minicom" = "y" ]; then
             echo ""
             echo -n "Detecting connection to board... "
             loopCount=0
-            serial_number=`lsusb -v -d 0403:6011 | grep iSerial | awk '{ print $NF }'`
+            serial_number=`sudo lsusb -v -d 0403:6011 | grep iSerial | awk '{ print $NF }'`
             usb_id=`dmesg | grep "SerialNumber: $serial_number" | tail -1 | awk '{ print $3 }'`
             port=`dmesg | grep FTDI | grep "tty" | grep "$usb_id" | tail -4 | head -1 | grep "attached" |  awk '{ print $NF }'`
             while [ -z "$port" ] && [ "$loopCount" -ne "10" ]
@@ -228,7 +228,7 @@ if [ "$minicom" = "y" ]; then
                 loopCount=$((loopCount+1))
 
                 sleep 1
-                serial_number=`lsusb -v -d 0403:6011 | grep iSerial | awk '{ print $NF }'`
+                serial_number=`sudo lsusb -v -d 0403:6011 | grep iSerial | awk '{ print $NF }'`
                 usb_id=`dmesg | grep "SerialNumber: $sn" | tail -1 | awk '{ print $3 }'`
                 port=`dmesg | grep FTDI | grep "tty" | grep "usb $usb_id" | tail -4 | head -1 | grep "attached" |  awk '{ print $NF }'`
             done
-- 
2.17.1

