From b4b329bf680ba31ad6f8959fdb962f81f457ef85 Mon Sep 17 00:00:00 2001
From: Hao Zhang <hzhang@ti.com>
Date: Mon, 21 Dec 2015 09:51:52 -0500
Subject: [PATCH 2/2] edma3: k2h: fix Linux user space drv/rm libs build error

Signed-off-by: Hao Zhang <hzhang@ti.com>
---
 packages/component.mk |    6 ------
 packages/makefile     |   12 ++++++------
 2 files changed, 6 insertions(+), 12 deletions(-)

diff --git a/packages/component.mk b/packages/component.mk
index dce1439..f243381 100755
--- a/packages/component.mk
+++ b/packages/component.mk
@@ -127,12 +127,6 @@ am571x-evm_a15_format_support = ELF
 am571x-evm_m4_format_support = ELF
 am571x-evm_66_format_support = ELF
 
-tci6636k2h-evm_supported_targets = a15 66
-tci6636k2h-evm_a15_cores = a15host
-tci6636k2h-evm_66_cores = c6xdsp
-tci6636k2h-evm_a15_format_support = ELF
-tci6636k2h-evm_66_format_support = ELF
-
 #Prepare library list to build from the PLATFORM and TARGET
 ifeq ($(PLATFORM),generic)
 edma3_lld_LIBS_ALL = edma3_lld_rm_generic
diff --git a/packages/makefile b/packages/makefile
index ba7d450..e8c3701 100755
--- a/packages/makefile
+++ b/packages/makefile
@@ -2404,9 +2404,9 @@ endif
 
 edma3_lld_tci6636k2h-evm_a15_libs_drvsample:
 ifeq ($(FORMAT),ELF)
-ifneq ($(TOOLCHAIN_a15),GCC)
-#	$(ECHO) \# Not supported a15:debug:edma3_lld_drv_sample
-#else
+ifeq ($(TOOLCHAIN_a15),GCC)
+	$(ECHO) \# Not supported user space a15:debug:edma3_lld_drv_sample
+else
 	$(ECHO) \# Making a15:debug:edma3_lld_drv_sample
 	$(MAKE) -C $(edma3_lld_drv_sample_PATH) PLATFORM=tci6636k2h-evm CORE=a15host PROFILE_a15host=debug
 	$(ECHO) \# Making a15:release:edma3_lld_drv_sample
@@ -2416,9 +2416,9 @@ endif
 
 edma3_lld_tci6636k2h-evm_a15_libs_rmsample:
 ifeq ($(FORMAT),ELF)
-ifneq ($(TOOLCHAIN_a15),GCC)
-#	$(ECHO) \# Not supported a15:debug:edma3_lld_rm_sample
-#else
+ifeq ($(TOOLCHAIN_a15),GCC)
+	$(ECHO) \# Not supported user space a15:debug:edma3_lld_rm_sample
+else
 	$(ECHO) \# Making a15:debug:edma3_lld_rm_sample
 	$(MAKE) -C $(edma3_lld_rm_sample_PATH) PLATFORM=tci6636k2h-evm CORE=a15host PROFILE_a15host=debug
 	$(ECHO) \# Making a15:release:edma3_lld_rm_sample
-- 
1.7.9.5

