From 094d58632315b9f9d2892d67d35d2121f5323d62 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Tue, 14 Feb 2017 15:00:34 -0500
Subject: [PATCH 10/86] hsr: add tx stats for master interface

Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 net/hsr/hsr_device.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/hsr/hsr_device.c b/net/hsr/hsr_device.c
index 517c699..850d744 100644
--- a/net/hsr/hsr_device.c
+++ b/net/hsr/hsr_device.c
@@ -241,6 +241,8 @@ static int hsr_dev_xmit(struct sk_buff *skb, struct net_device *dev)
 	master = hsr_port_get_hsr(hsr, HSR_PT_MASTER);
 	skb->dev = master->dev;
 	hsr_forward_skb(skb, master);
+	master->dev->stats.tx_packets++;
+	master->dev->stats.tx_bytes += skb->len;
 
 	return NETDEV_TX_OK;
 }
-- 
1.9.1

