From 17392e6e5f5c8f7b19f9b3d220e0a89a56d2f1c1 Mon Sep 17 00:00:00 2001
From: Eric Ruei <e-ruei1@ti.com>
Date: Wed, 13 Dec 2017 11:36:00 -0500
Subject: [PATCH 126/127] drivers: cpts: enable 1pps features per device tree

Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 drivers/net/ethernet/ti/cpts.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/ti/cpts.c b/drivers/net/ethernet/ti/cpts.c
index 73e24f1..4b03924 100644
--- a/drivers/net/ethernet/ti/cpts.c
+++ b/drivers/net/ethernet/ti/cpts.c
@@ -635,10 +635,9 @@ static long cpts_overflow_check(struct ptp_clock_info *ptp)
 	.owner		= THIS_MODULE,
 	.name		= "CTPS timer",
 	.max_adj	= 1000000,
-	.n_ext_ts	= CPTS_MAX_LATCH,
+	.n_ext_ts	= 0,
 	.n_pins		= 0,
-	.n_per_out	= 1,
-	.pps		= 1,
+	.pps		= 0,
 	.adjfreq	= cpts_ptp_adjfreq,
 	.adjtime	= cpts_ptp_adjtime,
 	.gettime64	= cpts_ptp_gettime,
@@ -1068,6 +1067,11 @@ struct cpts *cpts_create(struct device *dev, void __iomem *regs,
 				ret);
 			return ERR_PTR(ret);
 		}
+
+		/* Enable 1PPS related features	*/
+		cpts->info.pps		= 1;
+		cpts->info.n_ext_ts	= CPTS_MAX_LATCH;
+		cpts->info.n_per_out	= 1;
 	}
 
 	return cpts;
-- 
1.9.1

