From 68f47d036ee250b617c3b0af7b16dd9db988a4ea Mon Sep 17 00:00:00 2001
From: Djordje Senicic <x0157990@ti.com>
Date: Wed, 28 Feb 2018 11:44:03 -0500
Subject: [PATCH 187/189] connectivity: Add FTDI_SIO loadable module to
 configuration

Signed-off-by: Djordje Senicic <x0157990@ti.com>
---
 ti_config_fragments/connectivity.cfg | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ti_config_fragments/connectivity.cfg b/ti_config_fragments/connectivity.cfg
index 92d6dc1..ead85bb 100644
--- a/ti_config_fragments/connectivity.cfg
+++ b/ti_config_fragments/connectivity.cfg
@@ -357,6 +357,7 @@ CONFIG_USB_TEST=m
 CONFIG_USB_ACM=m
 CONFIG_USB_SERIAL=m
 CONFIG_USB_FTDI_SIO=m
+CONFIG_USB_SERIAL_FTDI_SIO=m
 
 # Extcon drivers
 CONFIG_EXTCON=m
-- 
1.9.1

