From f1893eab56d1618b80f03e6b30793d86a51e1fd1 Mon Sep 17 00:00:00 2001
From: Djordje Senicic <d-senicic1@ti.com>
Date: Mon, 12 Dec 2016 19:42:31 -0500
Subject: [PATCH] Fix abosulute path of include directory

Signed-off-by: Djordje Senicic <d-senicic1@ti.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ba110c..f16741b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   SET(COMMON_LIBS_PRIVATE "")
   set(COMMON_INCLUDE
     ${COMMON_INCLUDE}
-    /usr/include/libusb-1.0
+    ${VOXELSDK_INCDIR}/libusb-1.0
     Util/
   )
   
-- 
1.9.1

