diff --git a/build/buildlib.xs b/build/buildlib.xs
index e5e8ffb..5413530 100644
--- a/build/buildlib.xs
+++ b/build/buildlib.xs
@@ -559,24 +559,28 @@ function buildLibrary (libOptions, libName, evmName, target, libFiles)
     /* Create the Epilogue; which executes after all the builds are completed. 
      * This is used to generate the benchmark information for the built library. 
      * Also add the benchmarking information file to the package. */
-    Pkg.makeEpilogue += ".libraries: benchmarking_" + evmName + "_" + libName + "_" + target.suffix + "\n";
-    Pkg.makeEpilogue += "benchmarking_" + evmName + "_" + libName + "_" + target.suffix + ":";
+/* Putting a temp file in the object directory because javascript doesnt have a tempfile and we dont want collisions in multiple jobs*/ 
+ var tempFile = "$(LIBDIR)/" + evmName + "/" + targetDir + "/obj/tmp_" + target.suffix + ".xml";
+    // + evmName + "_" + libName + "_" + target.suffix + "\n";
+Pkg.makeEpilogue += ".libraries: " + lldFullLibraryPath + ".a" + target.suffix + "_size.txt\n";
+    // + evmName + "_" + libName + "_" + target.suffix + ":";
+Pkg.makeEpilogue += lldFullLibraryPath + ".a" + target.suffix + "_size.txt" + ": " + lldFullLibraryPath + ".a" + target.suffix;
     if ( java.lang.String(target.name).contains('66') )
     {    
-        Pkg.makeEpilogue += "\n\t $(C6X_GEN_INSTALL_PATH)/bin/ofd6x -x " + lldFullLibraryPath + ".a" + target.suffix + " > tmp.xml";
-        Pkg.makeEpilogue += "\n\t sectti tmp.xml > " + lldFullLibraryPath + ".a" + target.suffix +  "_size.txt";
+        Pkg.makeEpilogue += "\n\t $(C6X_GEN_INSTALL_PATH)/bin/ofd6x -x " + lldFullLibraryPath + ".a" + target.suffix + " > " + tempFile;
+        Pkg.makeEpilogue += "\n\t $(SECTTI) " + tempFile + " > " + lldFullLibraryPath + ".a" + target.suffix +  "_size.txt";
     }   
     else if (target.name == "M4")
     {
-        Pkg.makeEpilogue += "\n\t $(TOOLCHAIN_PATH_M4)/bin/armofd -x " + lldFullLibraryPath + ".a" + target.suffix + " > tmp.xml";
-        Pkg.makeEpilogue += "\n\t sectti tmp.xml > " + lldFullLibraryPath + ".a" + target.suffix +  "_size.txt";
+        Pkg.makeEpilogue += "\n\t $(TOOLCHAIN_PATH_M4)/bin/armofd -x " + lldFullLibraryPath + ".a" + target.suffix + " > " + tempFile;
+        Pkg.makeEpilogue += "\n\t $(SECTTI) " + tempFile + " > " + lldFullLibraryPath + ".a" + target.suffix +  "_size.txt";
     }
     else
     {
-        Pkg.makeEpilogue += "\n\t $(TOOLCHAIN_PATH_A15)/bin/$(CROSS_TOOL_PRFX)size " + lldFullLibraryPath + ".a" + target.suffix + " > tmp.xml";
-        Pkg.makeEpilogue += "\n\t $(MV) tmp.xml " + lldFullLibraryPath + ".a" + target.suffix +  "_size.txt";
+        Pkg.makeEpilogue += "\n\t $(TOOLCHAIN_PATH_A15)/bin/$(CROSS_TOOL_PRFX)size " + lldFullLibraryPath + ".a" + target.suffix + " > " + tempFile;
+        Pkg.makeEpilogue += "\n\t $(MV) " + tempFile + " " + lldFullLibraryPath + ".a" + target.suffix +  "_size.txt";
     }                
-    Pkg.makeEpilogue += "\n\t $(RM) tmp.xml\n\n";
+    Pkg.makeEpilogue += "\n\t $(RM) " + tempFile + "\n\n";
     Pkg.otherFiles[Pkg.otherFiles.length++] = lldFullLibraryPath + ".a" + target.suffix + "_size.txt";
     Pkg.otherFiles[Pkg.otherFiles.length++] = lldFullBuildPath + ".a" + target.suffix + ".mk";
     Pkg.otherFiles[Pkg.otherFiles.length++] = lldFullLibraryPath + ".a" + target.suffix;
diff --git a/docs/Module.xs b/docs/Module.xs
index 4763fee..2221c64 100644
--- a/docs/Module.xs
+++ b/docs/Module.xs
@@ -21,7 +21,9 @@ var libUtility = xdc.loadCapsule ("../build/buildlib.xs");
 function modBuild() 
 {
     /* Create the actual PROLOGUE Section for the Documentation.*/
+Pkg.makePrologue += ".PHONY: csl_document_generation\n";
     Pkg.makePrologue += "release: csl_document_generation\n";
+Pkg.makePrologue += "docs/doxygen docs/doxygen/html/index.html: csl_document_generation\n";
     Pkg.makePrologue += "csl_document_generation:\n";
     Pkg.makePrologue += "\t @echo ----------------------------\n";
     Pkg.makePrologue += "\t @echo Generating CSL Documentation\n";
