--- a/scripts/turtlebot_teleop_key
+++ b/scripts/turtlebot_teleop_key
@@ -103,8 +103,8 @@
     control_speed = 0
     control_turn = 0
     try:
-        print msg
-        print vels(speed,turn)
+        print (msg)
+        print (vels(speed,turn))
         while(1):
             key = getKey()
             if key in moveBindings.keys():
@@ -116,9 +116,9 @@
                 turn = turn * speedBindings[key][1]
                 count = 0
 
-                print vels(speed,turn)
+                print (vels(speed,turn))
                 if (status == 14):
-                    print msg
+                    print (msg)
                 status = (status + 1) % 15
             elif key == ' ' or key == 'k' :
                 x = 0
@@ -160,7 +160,7 @@
             #print("publihsed: vx: {0}, wz: {1}".format(twist.linear.x, twist.angular.z))
 
     except:
-        print e
+        print (e)
 
     finally:
         twist = Twist()
