--- a/launch/minimal.launch
+++ b/launch/minimal.launch
@@ -1,11 +1,10 @@
 <launch>
   <!-- Turtlebot -->
   <arg name="base"              default="kobuki"         doc="mobile base type [create, roomba]"/>
-  <arg name="battery"           default="$(env TURTLEBOT_BATTERY)"      doc="kernel provided locatio for battery info, use /proc/acpi/battery/BAT0 in 2.6 or earlier kernels." />
-  <arg name="stacks"            default="$(env TURTLEBOT_STACKS)"       doc="stack type displayed in visualisation/simulation [circles, hexagons]"/>
-  <arg name="3d_sensor"         default="radar"    doc="3d sensor types [kinect, asux_xtion_pro, radar]"/>
-  <arg name="simulation"        default="$(env TURTLEBOT_SIMULATION)"   doc="set flags to indicate this turtle is run in simulation mode."/>
-  <arg name="serialport"        default="$(env TURTLEBOT_SERIAL_PORT)"  doc="used by create to configure the port it is connected on [/dev/ttyUSB0, /dev/ttyS0]"/>
+  <arg name="stacks"            default="hexagons"       doc="stack type displayed in visualisation/simulation [circles, hexagons]"/>
+  <arg name="3d_sensor"         default="radar"          doc="3d sensor types [kinect, asux_xtion_pro, radar]"/>
+  <arg name="simulation"        default="false"          doc="set flags to indicate this turtle is run in simulation mode."/>
+  <arg name="serialport"        default="/dev/kobuki"    doc="used by create to configure the port it is connected on [/dev/ttyUSB0, /dev/ttyS0]"/>
 
   <param name="/use_sim_time" value="$(arg simulation)"/>
 
@@ -18,7 +17,4 @@
     <arg name="base" value="$(arg base)" />
     <arg name="serialport" value="$(arg serialport)" />
   </include>
-  <include unless="$(eval arg('battery') == 'None')" file="$(find turtlebot_bringup)/launch/includes/netbook.launch.xml">
-    <arg name="battery" value="$(arg battery)" />
-  </include>
 </launch>
